### PR TITLE
MCR-2826 pandoc module

### PR DESCRIPTION
**Pull request by chemicus**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2826).

### CODE REVIEW

File: mycore-pandoc/src/main/java/org/mycore/pandoc/MCRPandocAPI.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    private static String convertFormatToPath(Action action, String format) {
+        String property = MCRConfiguration2.getString("MCR.Pandoc." + action + "." + format).orElse("");
+        String path = MCRConfiguration2.getString("MCR.Pandoc." + action + "." + format + ".Path").orElse("");
+        if(!property.isEmpty()) {
+            if(path.isEmpty()) {
+                return Thread.currentThread().getContextClassLoader().getResource(property).getFile();
+            } else {
+                return Paths.get(path).resolve(property).toString();
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/java/org/mycore/pandoc/MCRPandocAPI.java:115]: @sonatype-lift ignore

File: mycore-pandoc/src/main/java/org/mycore/pandoc/MCRPandocAPI.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            }
+        } else {
+            return format;
+        }
+    }
+
+    private static byte[] callPandoc(String[] args, byte[] input) throws MCRPandocException {
+        ProcessBuilder pb = new ProcessBuilder(args);
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/java/org/mycore/pandoc/MCRPandocAPI.java:123]: @sonatype-lift ignore

File: mycore-pandoc/src/main/resources/lua/general.lua [hunk 3]
```diff
@@ -0,0 +1,54 @@
+-- This function is called once for the whole document. Parameters:
+-- body is a string, metadata is a table, variables is a table.
+function Doc(body, metadata, variables)
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua]: @sonatype-lift ignoreall

[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua]: @sonatype-lift ignoreall

File: mycore-pandoc/src/main/resources/lua/general.lua [hunk 4]
```diff
@@ -0,0 +1,54 @@
+-- This function is called once for the whole document. Parameters:
+-- body is a string, metadata is a table, variables is a table.
+function Doc(body, metadata, variables)
+    print(transform(metadata))
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua]: @sonatype-lift ignoreall

File: mycore-pandoc/src/main/resources/lua/general.lua [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+         s = s .. '['..k..'] = ' .. dumpPandocAst(v) .. ',\n'
+      end
+      return s .. '} '
+   else
+      return tostring(o)
+   end
+end
+
+meta = {}
+meta.__index =
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua:32]: @sonatype-lift ignore

File: mycore-pandoc/src/main/resources/lua/general.lua [hunk 6]
```diff
@@ -0,0 +1,36 @@
+stringify = pandoc.utils.stringify
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua:1]: @sonatype-lift ignoreall

File: mycore-pandoc/src/main/resources/lua/general.lua [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        end
+        return s .. '} '
+    else
+        return stringify(o)
+    end
+end
+
+meta = {}
+meta.__index =
+function(_, key)
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua:33]: @sonatype-lift ignore

File: mycore-pandoc/src/main/resources/lua/general.lua [hunk 8]
```diff
@@ -0,0 +1,36 @@
+stringify = pandoc.utils.stringify
+
+-- This function is called once for the whole document. Parameters:
+-- doc is the pandoc AST, doc.meta is a table of metadata, opts contains writer options
+-- cf. https://pandoc.org/custom-writers.html
+function Writer (doc, opts)
```
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua:6]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua:6]: @sonatype-lift ignore

File: mycore-pandoc/src/main/resources/lua/general.lua (hunk 6 above)
[the remark this answers is not in the document]
Comment (chemicus) [reply mycore-pandoc/src/main/resources/lua/general.lua:1]: @sonatype-lift ignore

**Review by kkrebs — APPROVED** [review]
presented at devcon 03/2023 and approved from community

**Pull request merged**